Doc/c-api/init.rst

```diff
@@ -855,6 +855,12 @@ code, or when embedding the Python interpreter:
    created, the current thread must not have acquired it, otherwise deadlock
    ensues.
 
+   .. warning::
+      Calling this function from a thread when the interpreter is finalizing
+      will terminate the thread, even if the thread was not created by Python.
+      You can use :c:func:`_Py_IsFinalizing` or :func:`sys.is_finalizing` to
+      check if the interpreter is in process of being finalized before calling
+      this function to avoid unwanted termination.
```

Comment:
You cannot really avoid it. If you call one of those functions, it means you want to call the C API afterwards...

Reply:
Not really, the application can select to avoid calling into Python because the interpreter is shutting down to avoid being killed. This is also specially important if the interpreter shutdown does not mean application shutdown, like when Python is embedded.